bifromq-inbox/bifromq-inbox-client/src/main/java/com/baidu/bifromq/inbox/client/InboxClient.java

```diff
@@ -58,12 +66,15 @@ final class InboxClient implements IInboxClient {
         fetchPipelineCache = Caffeine.newBuilder()
             .weakValues()
             .executor(MoreExecutors.directExecutor())
-            .removalListener((RemovalListener<FetchPipelineKey, InboxFetchPipeline>) (key, value, cause) -> {
-                if (value != null) {
-                    value.close();
-                }
-            })
-            .build(key -> new InboxFetchPipeline(key.tenantId, key.delivererKey, rpcClient));
+            .build(key -> {
+                IMessageStream<InboxFetched, InboxFetchHint> messageStream =
```

Comment:
1. why coupling cleaner logic with cache building logic?
2. add UT to assert the cleaner is work as expected

Reply:
fixed